pkg/agent/handler/agentresource.go

```diff
@@ -0,0 +1,39 @@
+package handler
+
+import (
+	"github.com/Axway/agent-sdk/pkg/agent/resource"
+	v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+	"github.com/Axway/agent-sdk/pkg/watchmanager/proto"
+)
+
+const (
+	discoveryAgent    = "DiscoveryAgent"
+	traceabilityAgent = "TraceabilityAgent"
+	governanceAgent   = "GovernanceAgent"
+)
+
+type agentResourceHandler struct {
+	agentResourceManager resource.Manager
+}
+
+// NewAgentResourceHandler - creates a Handler for Agent resources
+func NewAgentResourceHandler(agentResourceManager resource.Manager) Handler {
+	return &agentResourceHandler{
+		agentResourceManager: agentResourceManager,
+	}
+}
+
+func (h *agentResourceHandler) Handle(action proto.Event_Type, _ *proto.EventMeta, resource *v1.ResourceInstance) error {
+	if h.agentResourceManager != nil && action == proto.Event_UPDATED {
+		kind := resource.Kind
+		switch kind {
+		case discoveryAgent:
+			fallthrough
+		case traceabilityAgent:
+			fallthrough
+		case governanceAgent:
+			h.agentResourceManager.SetAgentResource(resource)
```

Comment:
i guess same here... if SetAgentResource can return error, it would get eaten by the return nil